blockly-core/core/ui/block.js

```diff
@@ -1796,7 +1809,7 @@ Blockly.Block.prototype.setNextStatement = function(hasNext, opt_check) {
     this.nextConnection.dispose();
     this.nextConnection = null;
   }
-  if (hasNext) {
+  if (hasNext && !this.nextConnectionDisabled_) {
```

Comment:
If you agree, maybe worth noting in comments that this won't prevent `setNextStatement(true)` from succeeding when called from block definitions.